ast/dml.go

```diff
@@ -166,6 +229,29 @@ type IndexHint struct {
 	HintScope  IndexHintScope
 }
 
+func (n *IndexHint) Restore(sb *strings.Builder) error {
+
+	indexHintType := indexHintTypeMap[int(n.HintType)]
+
+	indexHintScope := indexHintScopeMap[int(n.HintScope)]
+
+	tempIndexNames := []string{}
+	for _, value := range n.IndexNames {
+		tempIndexNames = append(tempIndexNames, value.String())
+	}
+	indexNames := "(" + strings.Join(tempIndexNames, ",") + ")"
+
+	result := []string{
+		indexHintType,
+		indexHintScope,
+		indexNames,
+	}
+	sb.WriteString(strings.Join(ListFilterTrim(result), " "))
```

Comment:
You could just build the `sb` incrementally

```go
sb.WriteString(indexHintType)
if len(indexHintScope) > 0 {
    sb.WriteString(" ")
    sb.WriteString(indexHintScope)
}
sb.WriteString(" (")
for i, value := range n.IndexNames {
    if i > 0 {
        sb.WriteString(", ")
    }
    WriteName(sb, value.O)
}
sb.WriteString(")")
```

Reply:
thank you, I think it is complicated, I will modify it now.